ruby/inc/leatherman/ruby/api.hpp

```diff
@@ -0,0 +1,664 @@
+        // TODO: What do we want to do with it?
+//        /**
+//         * Converts the given value to a corresponding Ruby object.
+//         * @param val The value to convert.
+//         * @return Returns a Ruby object for the value.
+//         */
+//        VALUE to_ruby(facter::facts::value const* val) const;
+
+        /**
+         * Looks up a constant based on the given names. The individual entries correspond to
+         * namespaces, as in {"A", "B", "C"} => A::B::C.
+         * @param names The names to lookup.
+         * @return Returns the value or raises a NameError.
+         */
+        VALUE lookup(std::initializer_list<std::string> const& names) const;
+
+        /**
+         * Determines if two values are equal.
+         * @param first The first value to compare.
+         * @param second The second value to compare.
+         * @return Returns true if eql? returns true for the first and second values.
+         */
+        bool equals(VALUE first, VALUE second) const;
+
+        /**
+         * Determines if the first value has case equality (===) with the second value.
+         * @param first The first value to compare.
+         * @param second The second value to compare.
+         * @return Returns true if === returns true for the first and second values.
+         */
+        bool case_equals(VALUE first, VALUE second) const;
+
+        /**
+         * Evalutes a string as ruby code.
+         * Any exception raised will be propagated as a C++ runtime_error
+         * @param code the ruby code to execute
+         */
+        void eval(const std::string& code);
+
+        /**
+         * Gets the underlying native instance from a Ruby data object.
+         * The Ruby object must have been allocated with rb_data_object_alloc.
+         * @tparam T The underlying native type.
+         * @param obj The Ruby data object to get the native instance for.
+         * @return Returns a pointer to the underlying native type.
+         */
+        template <typename T>
+        T* to_native(VALUE obj) const
+        {
+            return reinterpret_cast<T*>(reinterpret_cast<RData*>(obj)->data);
+        }
+
+        /**
+         * Registers a data object for cleanup when the API is destructed.
+         * The object must have been created with rb_data_object_alloc.
+         * @param obj The data object to register.
+         */
+        void register_data_object(VALUE obj) const
+        {
+            _data_objects.insert(obj);
+        }
+
+        /**
+         * Unregisters a data object.
+         * @param obj The data object to unregister.
+         */
+        void unregister_data_object(VALUE obj) const
+        {
+            _data_objects.erase(obj);
+        }
+
+        /**
+         * Global flag to disable Ruby VM cleanup.
+         * This should be set to false in any forked child processes.
+         */
+        static bool cleanup;
```

Comment:
I think this was removed, in https://github.com/puppetlabs/facter/commit/78be9ea963ec3fd24ecbf3ef2ba038c4b33b65b2#diff-fabbc6ab7f9f6a77c8cf164caa3860b6.